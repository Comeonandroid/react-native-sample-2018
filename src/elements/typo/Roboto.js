import React, { PureComponent } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { prop } from 'styled-tools'

class RobotoTextComponent extends PureComponent {
  render () {
    return (
      <RobotoText>{ this.props.children }</RobotoText>
    )
  }
}

export const RobotoText = styled.Text`
  font-family: ${ prop('theme.typo.condensedFont') };
  font-weight: 400;
`

RobotoTextComponent.propTypes = {
  children: PropTypes.node
}

export default RobotoText
