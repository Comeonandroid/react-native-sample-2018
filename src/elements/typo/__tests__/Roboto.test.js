import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import Roboto from '../Roboto'

describe('<Roboto />', () => {
  itBehavesLike('aThemedComponent', { subject: <Roboto>Test</Roboto> })
})
