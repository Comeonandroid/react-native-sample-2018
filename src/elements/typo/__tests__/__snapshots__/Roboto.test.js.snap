// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Roboto /> dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "Roboto Condensed",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;

exports[`<Roboto /> light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "Roboto Condensed",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;
